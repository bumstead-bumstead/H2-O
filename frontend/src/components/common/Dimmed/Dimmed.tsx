import type { HTMLAttributes } from 'react';
import { css, keyframes } from '@emotion/react';
import styled from '@emotion/styled';

interface Props extends HTMLAttributes<HTMLDivElement> {
  enableAnimation?: boolean;
}

const Dimmed = styled.div<Props>`
  background-color: ${({ theme }) => theme.colors.dimmed};
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  backdrop-filter: blur(6px);
  z-index: 10;

  ${({ enableAnimation }) =>
    enableAnimation &&
    css`
      animation: ${dimmedAnimation} 1s ease;
    `}
`;

export default Dimmed;

const dimmedAnimation = keyframes`
  from {
    opacity: 0;
  }
  to {
    opacity: 1;

  }
`;
