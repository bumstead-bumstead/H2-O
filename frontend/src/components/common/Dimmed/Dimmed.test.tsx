import Dimmed from './Dimmed';
import { render, screen, userEvent, waitFor } from '@/tests/test-util';

describe('Dimmed 컴포넌트 테스트', () => {
  it('Dimmed가 정상적으로 렌더링된다.', () => {
    render(<Dimmed data-testid='dimmed' />);

    expect(screen.getByTestId('dimmed')).toBeInTheDocument();
  });

  it('onClick 이벤트가 정상적으로 동작한다.', async () => {
    const handleClick = vi.fn();
    render(<Dimmed data-testid='dimmed' onClick={handleClick} />);

    expect(handleClick).not.toBeCalled();
    userEvent.click(screen.getByTestId('dimmed'));

    await waitFor(() => expect(handleClick).toBeCalledTimes(1));
  });

  it('Dimmed 스냅샷 테스트', () => {
    render(<Dimmed data-testid='dimmed' />);

    expect(screen.getByTestId('dimmed')).toMatchSnapshot();
  });
});
