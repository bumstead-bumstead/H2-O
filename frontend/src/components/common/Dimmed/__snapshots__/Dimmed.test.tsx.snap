// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dimmed 컴포넌트 테스트 > Dimmed 스냅샷 테스트 1`] = `
.emotion-0 {
  background-color: #1F1F1FB3;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  -webkit-backdrop-filter: blur(6px);
  backdrop-filter: blur(6px);
  z-index: 10;
}

<div
  class="emotion-0"
  data-testid="dimmed"
/>
`;
